com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -311,6 +311,18 @@ public string path
             }
         }
 
+        Material m_Material;
```

Comment:
What is this material used for?   I would think we don't need to store the material in the source asset, just in the imported artifacts...

Reply:
Ah, I see, it's a non-serialized cache of the runtime generated material, to enable the instant update..

Reply:
Being part of graph data makes it potentially re-entrant, which is not ideal.   In the sense that it should not be used by any part of import as it's data from the previous import.  It's basically a short-circuit to modify the output results, in situations where we know the changes only modify the material.   I think it would be better to move this cache to another location in ShaderGraph...

Reply:
or at minimum name the public getter something scary, like "defaultMaterialArtifact" or something...

Reply:
And add a comment that this is to do immediate updates of the material, skipping the import job.

Reply:
Also:  how do we back out these changes if the user doesn't save the graph?

Reply:
I moved the property to the GraphObject, maybe it makes a bit more sense. Also renamed it to materialArtifact with comment.

You're right, I didn't think about discarding changes, it's fixed